tensorflow/lite/tools/benchmark/experimental/firebase/android/src/org/tensorflow/lite/benchmark/firebase/BenchmarkModelActivity.java

```diff
@@ -43,8 +46,27 @@ public void onCreate(Bundle savedInstanceState) {
       Log.e(TAG, "Received non Firebase Game Loop test intent " + intent.getAction());
       finish();
     }
-    int scenario = intent.getIntExtra("scenario", 0);
-    Log.i(TAG, "Running TensorFlow Lite benchmark with scenario: " + scenario);
+    String args = null;
+    try(BufferedReader reader = new BufferedReader(new FileReader("/data/local/tmp/input.txt"))) {
+      args = reader.readLine();
+    }
+    catch (IOException exception){
+      throw new RuntimeException("input.txt not found", exception);
+    }
+    File file = new File("/data/local/tmp/graph");
+    if(!file.exists()){
+      throw new RuntimeException("tflite does not exist");
+    }
+    if(args==null){
+      args="";
+    }
+
+    if (args.contains("--use_hexagon=true") || args.contains("--use_hexagon=1")) {
+      // Users should not specify this argument.
+      args = args + " --hexagon_lib_path=" + getApplicationInfo().nativeLibraryDir;
+    }
+    args += " --graph=/data/local/tmp/graph ";
+    Log.i(TAG, "Running TensorFlow Lite benchmark with args: " + args);
```

Comment:
Again, for your future self - I would just refactor it to a function "parse args" or something like that.

In a month you are gonna forget what this code does.